msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -137,6 +137,117 @@ async def on_message_activity(self, turn_context: TurnContext):
+
+* You can configure your bot to send a follow up message if your app user hasn't read the message in the personal chat.
+
+* You can create a feedback loop using read receipts to tune your bot’s experience.
+
+> [!NOTE]
+>
+> * Read receipt events are available in [public developer preview](~/resources/dev-preview/developer-preview-intro.md).
+> * Read receipts are supported only in user to bot chat scenarios.
+> * Read receipts for bots doesn’t support team, channel, and group chat scopes.
+> * If a tenant admin or user disables the **Read receipts** setting, the bot doesn't receive the read receipt event.
+
+To receive read receipts events for your bot, ensure the following:
+
+* Add the [RSC](~/graph-api/rsc/resource-specific-consent.md#rsc-permissions-for-a-chat-or-meeting) `ChatMessageReadReceipt.Read.Chat` permission in the [app manifest](~/resources/schema/manifest-schema.md), as follows: 
+
+    # [App manifest v1.12 or later](#tab/app-manifest-v112-or-later)
+    
+    ```json
+    
+    "webApplicationInfo": {
+    
+        "id": "38f0ca43-1c38-4c39-8097e-47f62c686500",
+        "resource": ""
+    },
+    "authorization": {
+        "permissions": {
+            "orgwide": [],
+            "resourceSpecific": [
+            {
+                "name": "ChatMessageReadReceipt.Read.Chat",
+                "type": "Application"
+            }
+            ]
+        }
+    }
+    
+    ```
+    
+    # [App manifest v1.11 or earlier](#tab/app-manifest-v111-or-earlier)
+    
+    ```json
+    
+     “webApplicationInfo”: {
+    
+         "id": "123456c8-67d2-4f54-b74e-408b195c4cbc",
+         "resource": "https: //AnyString",
+         "applicationPermissions": [
+             "ChatMessageReadReceipt.Read.Chat"
+         ]
+     },
+    
+    ```
+    
+    ---
+    
+    You can also add RSC permissions through Graph API. For more information, see [`consentedPermissionSet`](/graph/api/userteamwork-teamsappinstallation-upgrade#http-request).
+
+* Override the method `OnTeamsReadReceiptAsync` with `IsMessageRead` handler.
+
+  The `IsMessageRead` helper method is useful to determine if the message is read by the recipients. If the `compareMessageId` is less than or equal to the `LastReadMessageId`, then the message has been read. Override the `OnTeamsReadReceiptAsync` method to receive read receipts with [`IsMessageRead`](/dotnet/api/microsoft.bot.schema.teams.readreceiptinfo.ismessageread#microsoft-bot-schema-teams-readreceiptinfo-ismessageread(system-string)) helper method:
+
+    ```csharp
```

Comment:
Can we have a complete example for this using the IsMessageRead?

```
protected override async Task OnTeamsReadReceiptAsync(ReadReceiptInfo readReceiptInfo, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken) 
{
    var lastReadMessageId = readReceiptInfo.LastReadMessageId;
   if (IsMessageRead("{id of the message that you care}", LastReadMessageId))
   {
          await turnContext.SendActivityAsync(MessageFactory.Text("User read the bot's message"), cancellationToken);
    }
}
```

Reply:
Updated